test/Libraries/Microsoft.AspNetCore.HeaderParsing.Tests/HeaderParsingFeatureTests.cs

```diff
@@ -139,10 +139,11 @@ public void TryParse_returns_false_on_error()
 
         Assert.Equal("Can't parse header 'Date' due to 'Unable to parse date time offset value.'.", _logger.Collector.LatestRecord.Message);
 
-        var latest = metricCollector.GetCounterValues<long>(@"R9.HeaderParsing.ParsingErrors")!.LatestWritten!;
+        var latest = metricCollector.LatestMetricValueUpdated!;
         latest.Value.Should().Be(1);
-        latest.GetDimension("HeaderName").Should().Be("Date");
-        latest.GetDimension("Kind").Should().Be("Unable to parse date time offset value.");
+        Assert.True(latest.ContainsDimensions(
```

Comment:
Its totally up to you how you want to do these tests, but I'd expect this pattern makes the assert failures less informative. Instead of seeing something like "Expected value 'Date' but received actual value 'OtherDate'" now the assert will be a more vague "Assert failed, line 144".